### ISSUE TITLE
\htmlonly creates problems when used with XHTML. (Origin: bugzilla #657198)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#4377]
status NEW severity *normal* in component *general* for *---*
Reported in version *1.7.5.1* on platform *Other*
Assigned to: Dimitri van Heesch

Original attachment names and IDs:
- _[HTMLOnlyBug.zip](https://sourceforge.net/projects/doxygen/files/attachments/bug657198_194527_HTMLOnlyBug.zip/download/download)_ (ID 194527)
- _[Bug.dox](https://sourceforge.net/projects/doxygen/files/attachments/bug657198_195234_Bug.dox/download/download)_ (ID 195234)

On 2011-08-23 20:26:12 +0000, Bill wrote:

    Created attachment 194527
    Zip file containing doxygen config file and input file
    
    \htmlonly creates problems when used with XHTML.
    
    Output has two problems, unbalanced <p>'s and
    a generated code block that uses <iframe>:
    
        
        <div id="MSearchResultsWindow">
        <iframe src="javascript:void(0)" frameborder="0" 
                name="MSearchResults" id="MSearchResults">
        </iframe>
    
    Since <iframe> is not supported in XHTML, this is a problem.
    
    Includes are the Doxygen config file: showBug
    and a demo file: Bug.dox

On 2011-08-30 17:17:52 +0000, Bill wrote:

    Please ignore the second issue.  I am now aware that the "Transitional
    XHTML" supports iframes.
    
    To summarize, the issue is \htmlonly now breaks for XHTML output.
    I had been using \htmlonly to allow use of iframes under HTML.
    In spite of having balanced tags in the source file, the generated XHTML is
    not happy.
    I have attached a second source file that works for HTML, but fails for
    XHTML.

On 2011-08-30 17:19:29 +0000, Bill wrote:

    Created attachment 195234
    Example of iframe that works under HTML but fails under XHTML

**Comment by albert-github** [comment]
With version 1.7.5.1 I could reproduce the problems in the original example. In version 1.8.14 the problems are gone.

The second example contains non valid `xhtml` code:
```
<iframe  src="Notes.pdf" width="100%" height="1600" />
 <p>Your browser does not support iframes.</p>
</iframe>
```
should probably read
```
<iframe  src="Notes.pdf" width="100%" height="1600" >
 <p>Your browser does not support iframes.</p>
</iframe>

```
after this correction the problem runs for version 1.8.14

I think the problem has been solved in the mean time.
